### ISSUE TITLE
Exclusive ports

### ISSUE THREAD

**Issue by bapt** [freebsd/poudriere#87]
It would be nice to have an option to set some ports as "exclusive" or "CPU/memory intensive". Two ports with this flag shouldn't build in parallel.

For example using a new command line argument:
```
# poudriere bulk -x exclusive_ports_list.txt [..]
```

With content of exclusive_ports_list.txt being:
```
editors/libreoffice
www/chromium
www/firefox
```
..

<hr /><i>bdrewery added on 2013-03-11 11:55:27 UTC:</i><br />
This might be better as a consistent port option. Perhaps PACKAGE_BUILDING_MEMORY_INTENSIVE. Then we could mark all ports like this in the versioned ports tree. For instance, sometimes lang/pypy is memory intensive, sometimes it is not. The port could have specific logic to determine if it should advertise as intensive for that build.

Then poudriere would just serialize any of these ports.

<hr /><i>anonymous added on 2013-03-18 15:23:49 UTC:</i><br />
Although I agree for most it would be better to have a port option for this, what is memory/CPU intensive or not can be subjective sometimes and it may depend on the hardware used or in the user's point view more than in the port itself in some cases.

In any case, I leave the decision up to you, any of these methods would be great, thanks!

**Comment by grahamperrin** [comment]
If it will be in `poudriere.conf`, then maybe like this: 

`EXCLUSIVE_PORTS="llvm* www/chromium"`

If additionally (or alternatively) there will be separate file (or combination of files) to list the exclusivity, then could the file be **universal configuration library** (UCL) format? Usable with [devel/uclcmd](https://www.freshports.org/devel/uclcmd/) and so on. 

Generally: will support for globbing be reasonable?

Thank you

**Comment by bdrewery** [comment]
FWIW I have this done in a local branch for #822. It is named `MUTUALLY_EXCLUSIVE_BUILD_PACKAGES`. It depends on #822 as that incidentally reworked the queue to support the needs here.
